src/fstext/kaldi-fst-io.cc

```diff
@@ -0,0 +1,56 @@
+// fstext/kaldi-fst-io.cc
+
+// Copyright 2009-2011  Microsoft Corporation
+//           2012-2015  Johns Hopkins University (Author: Daniel Povey)
+//                2013  Guoguo Chen
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "fstext/kaldi-fst-io.h"
+#include "base/kaldi-error.h"
+#include "base/kaldi-math.h"
+#include "util/kaldi-io.h"
+
+namespace fst {
+
+VectorFst<StdArc> *ReadFstKaldi(std::string rxfilename) {
+  if (rxfilename == "") rxfilename = "-"; // interpret "" as stdin,
+  // for compatibility with OpenFst conventions.
+  kaldi::Input ki(rxfilename);
+  fst::FstHeader hdr;
+  if (!hdr.Read(ki.Stream(), rxfilename))
+    KALDI_ERR << "Reading FST: error reading FST header from "
+              << kaldi::PrintableRxfilename(rxfilename);
+  FstReadOptions ropts("<unspecified>", &hdr);
+  VectorFst<StdArc> *fst = VectorFst<StdArc>::Read(ki.Stream(), ropts);
+  if (!fst)
+    KALDI_ERR << "Could not read fst from "
+              << kaldi::PrintableRxfilename(rxfilename);
+  return fst;
+}
+
+void WriteFstKaldi(const VectorFst<StdArc> &fst,
+                          std::string wxfilename) {
```

Comment:
I'm addressing these comments but I won't reply individually to them.
Note: some of this code is very old, just moved around; but good to review
it anyway.

On Wed, Aug 19, 2015 at 6:51 PM, Kirill Katsnelson <notifications@github.com
> wrote:

> In src/fstext/kaldi-fst-io.cc
> <https://github.com/kaldi-asr/kaldi/pull/83#discussion_r37488774>:
>
> > +  // for compatibility with OpenFst conventions.
> > +  kaldi::Input ki(rxfilename);
> > +  fst::FstHeader hdr;
> > +  if (!hdr.Read(ki.Stream(), rxfilename))
> > +    KALDI_ERR << "Reading FST: error reading FST header from "
> > +              << kaldi::PrintableRxfilename(rxfilename);
> > +  FstReadOptions ropts("<unspecified>", &hdr);
> > +  VectorFst<StdArc> *fst = VectorFst<StdArc>::Read(ki.Stream(), ropts);
> > +  if (!fst)
> > +    KALDI_ERR << "Could not read fst from "
> > +              << kaldi::PrintableRxfilename(rxfilename);
> > +  return fst;
> > +}
> > +
> > +void WriteFstKaldi(const VectorFst<StdArc> &fst,
> > +                          std::string wxfilename) {
>
> Align to (.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/83/files#r37488774>.
>